tools/check_copyright_headers.sh

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+SCRIPT_DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
+SEARCH_DIR="$SCRIPT_DIR/.."
+
+COPYRIGHT_REGEX="Copyright[[:space:]]\\(c\\)[[:space:]]20+[0-9]{2}"
+# List of extensions to search for
+FILE_EXT=(
+    "*.rs"
+)
+# Exclude paths from search
+EXCLUDE_PATHS=(
+    "$SEARCH_DIR/target/*"
+)
```

Comment:
It's this part. The regex can fit with any file's copyright header, and in the future if we want more files checked and also we need to add more exclusion paths, we can do it easy.